cmd/zoekt-sourcegraph-indexserver/cleanup.go

```diff
@@ -71,6 +71,24 @@ func cleanup(indexDir string, repos []string, now time.Time) {
 
 		moveAll(trashDir, shards)
 	}
+
+	// Remove old .tmp files from crashed indexer runs-- for example, if
+	// an indexer OOMs, it will leave around .tmp files, usually in a loop.
+	maxAge := now.Add(-4 * time.Hour)
+	if failures, err := filepath.Glob(filepath.Join(indexDir, "*.tmp")); err != nil {
+		log.Printf("Glob: %v", err)
+	} else {
+		for _, f := range failures {
+			if st, err := os.Stat(f); err != nil {
+				log.Printf("Stat(%q): %v", f, err)
+			} else {
+				if !st.IsDir() && st.ModTime().Before(maxAge) {
+					log.Printf("removing old tmp file: %s", f)
+					os.Remove(f)
+				}
+			}
+		}
+	}
```

Comment:
```suggestion
	failures, err := filepath.Glob(filepath.Join(indexDir, "*.tmp")) 
	if err != nil {
		log.Printf("Glob: %v", err)
		return
	}
	for _, f := range failures {
		st, err := os.Stat(f)
		if err != nil {
			log.Printf("Stat(%q): %v", f, err)
			continue
		}
		if !st.IsDir() && st.ModTime().Before(maxAge) {
			log.Printf("removing old tmp file: %s", f)
			os.Remove(f)
		}
	}
```

Alternative way to structure control flow that reduces nesting/`else`s. I've tended to use this style thanks to many reviews from @unknwon :-). Up to you. If you want it though, don't apply this suggestion directly--I'm not sure the tabs are properly preserved.

Reply:
I was copying from https://github.com/sourcegraph/zoekt/blob/master/cmd/zoekt-indexserver/main.go#L157 , but I agree.

Reply:
I partially integrated this feedback. I didn't add an early return, but reduced the nesting in the for loop.